core/src/main/scala/tech/beshu/ror/boot/engines/ImpersonatorsReloadableEngine.scala

```diff
@@ -24,20 +24,40 @@ import tech.beshu.ror.RequestId
 import tech.beshu.ror.accesscontrol.domain.RorConfigurationIndex
 import tech.beshu.ror.boot.ReadonlyRest
 import tech.beshu.ror.boot.ReadonlyRest.StartingFailure
-import tech.beshu.ror.boot.RorInstance.RawConfigReloadError
+import tech.beshu.ror.boot.RorInstance.{RawConfigReloadError, TestConfig}
+import tech.beshu.ror.boot.engines.BaseReloadableEngine.EngineState
 import tech.beshu.ror.boot.engines.ConfigHash._
 import tech.beshu.ror.configuration.RawRorConfig
 
+import java.time.Clock
 import scala.concurrent.duration.FiniteDuration
 
 private[boot] class ImpersonatorsReloadableEngine(boot: ReadonlyRest,
                                                   reloadInProgress: Semaphore[Task],
                                                   rorConfigurationIndex: RorConfigurationIndex)
-                                                 (implicit scheduler: Scheduler)
+                                                 (implicit scheduler: Scheduler,
+                                                  clock: Clock)
   extends BaseReloadableEngine(
     "test", boot, None, reloadInProgress, rorConfigurationIndex
   ) {
 
+  def currentTestConfig()
+                       (implicit requestId: RequestId): Task[TestConfig] = {
+    Task.delay {
+      currentEngineState match {
+        case EngineState.NotStartedYet(None) =>
```

Comment:
maybe we can write it as follows:
`case EngineState.NotStartedYet(None) | EngineState.Stopped =>`